File: src/commands/server_side/RemoveHeadCommand.java
package commands.server_side;

import OrganizationDatabase.OrganizationManagerInterface;
import lib.ExecutionStatus;
import network.client.udp.User;

public class RemoveHeadCommand extends ServerSideCommand {

    public RemoveHeadCommand(ServerCallbackFunction callback) {
        super(callback);
    }

    @Override
    protected void executeImplementation(
            User user,
            OrganizationManagerInterface organizationManager,
            Object[] args,
            ServerCallbackFunction callback
    ) throws Exception {
        assert args.length == 0;

        callback.invoke(
                user,
                organizationManager,
                ExecutionStatus.SUCCESS,
                null,
                organizationManager.removeHead()
        );
    }
}

File: src/commands/server_side/RemoveHeadCommand.kt
package commands.server_side

import database.Organization
import database.OrganizationManagerInterface
import network.client.udp.User

class RemoveHeadCommand : ServerSideCommand {
    override suspend fun executeImplementation(
        user: User?,
        organizationManager: OrganizationManagerInterface,
        argument: Any?,
    ): Result<Organization?> {
        assert(argument == null)
        return Result.success(organizationManager.removeHead());
    }
}
